ush/setup.py

```diff
@@ -527,81 +527,32 @@ def get_location(xcs,fmt):
+            cycle hour.
               DATE_FIRST_CYCL = \"{DATE_FIRST_CYCL}\"'''
         )
 
     if not isinstance(DATE_LAST_CYCL, datetime.date):
         print_err_msg_exit(
             f'''
             DATE_LAST_CYCL must be a string consisting of exactly 8 digits of the
-            form \"YYYYMMDD\", where YYYY is the 4-digit year, MM is the 2-digit
-            month, and DD is the 2-digit day-of-month.
+            form \"YYYYMMDDHH\", where YYYY is the 4-digit year, MM is the 2-digit
+            month, DD is the 2-digit day-of-month, and HH is the 2-digit
+            cycle hour.
               DATE_LAST_CYCL = \"{DATE_LAST_CYCL}\"'''
         )
     #
     # -----------------------------------------------------------------------
     #
-    # Check that all elements of CYCL_HRS are strings consisting of exactly
-    # 2 digits that are between "00" and "23", inclusive.
-    #
-    # -----------------------------------------------------------------------
-    #
-    i = 0
-    for CYCL in CYCL_HRS:
-        if CYCL < 0 or CYCL > 23:
-            print_err_msg_exit(
-                f'''
-                Each element of CYCL_HRS must be an integer between \"00\" and \"23\", in-
-                clusive (including a leading \"0\", if necessary), specifying an hour-of-
-                day.  Element #{i} of CYCL_HRS (where the index of the first element is 0)
-                does not have this form:
-                  CYCL_HRS = {CYCL_HRS}
-                  CYCL_HRS[{i}] = \"{CYCL_HRS[i]}\"'''
-            )
-
-        i = i + 1
-    #
-    # -----------------------------------------------------------------------
-    # Check cycle increment for cycle frequency (cycl_freq).
-    # only if INCR_CYCL_FREQ < 24.
-    # -----------------------------------------------------------------------
-    #
-    if INCR_CYCL_FREQ < 24 and i > 1:
-        cycl_intv = 24 // i
-        if cycl_intv != INCR_CYCL_FREQ:
-            print_err_msg_exit(
-                f"""
-                The number of CYCL_HRS does not match with that expected by INCR_CYCL_FREQ:
-                  INCR_CYCL_FREQ = {INCR_CYCL_FREQ}
-                  cycle interval by the number of CYCL_HRS = {cycl_intv}
-                  CYCL_HRS = {CYCL_HRS} """
-            )
-
-        for itmp in range(1, i):
-            itm1 = itmp - 1
-            cycl_next_itmp = CYCL_HRS[itm1] + INCR_CYCL_FREQ
-            if cycl_next_itmp != CYCL_HRS[itmp]:
-                print_err_msg_exit(
-                    f'''
-                    Element {itmp} of CYCL_HRS does not match with the increment of cycle
-                    frequency INCR_CYCL_FREQ:
-                      CYCL_HRS = {CYCL_HRS}
-                      INCR_CYCL_FREQ = {INCR_CYCL_FREQ}
-                      CYCL_HRS[{itmp}] = \"{CYCL_HRS[itmp]}\"'''
-                )
-    #
-    # -----------------------------------------------------------------------
```

Comment:
This test didn't seem necessary any more. Please let me know if there is a better one to use here.

Reply:
Nice simplification!